packages/medusa/src/services/product.ts

```diff
@@ -369,62 +406,54 @@ class ProductService extends BaseService {
     }
 
     return priceIndex > -1
-      ? await this.setAdditionalPrices(
-          product,
-          config.currency_code,
-          config.region_id,
-          config.cart_id,
-          config.customer_id,
-          config.include_discount_prices
-        )
+      ? await this.setAdditionalPrices(product, {
+          cart_id: config.cart_id,
+          currency_code: config.currency_code,
+          customer_id: config.customer_id,
+          include_discount_prices: config.include_discount_prices,
+          region_id: config.region_id,
+        })
       : product
   }
 
   /**
    * Gets all variants belonging to a product.
-   * @param {string} productId - the id of the product to get variants from.
-   * @param {FindConfig<Product>} config - The config to select and configure relations etc...
-   * @return {Promise} an array of variants
+   * @param productId - the id of the product to get variants from.
+   * @param config - The config to select and configure relations etc...
+   * @return an array of variants
    */
   async retrieveVariants(
-    productId,
-    config = {
+    productId: string,
+    config: FindProductConfig = {
       skip: 0,
       take: 50,
-      relations: defaultAdminProductsVariantsRelations,
     }
-  ) {
+  ): Promise<ProductVariant[]> {
     const product = await this.retrieve(productId, config)
```

Comment:
done! btw, I'm a bit curious about this since this [change](https://github.com/medusajs/medusa/pull/1471/files#diff-ed73bfb406c02699ffb37440f8de2b7d34d9fdbc1942898c8f3a8def29a4a78c) was introduced a few weeks ago and removed the "variants" relation. Any rationale behind it? Also, when I was refactoring, one thing I found quite interesting was that no matter where I looked, I couldn't find any variable exported called `defaultAdminProductsVariantsRelations`, yet it was being used in the `ProductService` [here](https://github.com/medusajs/medusa/blob/c6f1de685f2189ea21d30a3ce31180d8cf3acd45/packages/medusa/src/services/product.js#L5) 😅